### PR TITLE
add Terminology section

### LINKED ISSUE #21: Add Terminology section
Add concepts for significant units of information that are referenced throughout this specification as well as that should be referenced from other documents/specifications.

The references to these concepts in this document can link back, e.g., `treat the <a href="#solid-profile">Solid Profile</a> as a graph`.

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
Closes #21 

Also removed some minor "to-do" comments that have been addressed. 

[Preview](https://htmlpreview.github.io/?https://github.com/solid/webid-profile/blob/9c28893b5652f3b08edf1929beca003e0d641652/index.html) | [Diff](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fsolid.github.io%2Fwebid-profile%2F&doc2=https%3A%2F%2Fraw.githubusercontent.com%2Fsolid%2Fwebid-profile%2F9c28893b5652f3b08edf1929beca003e0d641652%2Findex.html)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -972,7 +964,7 @@ <h3 property="schema:name skos:prefLabel">Terminology</h3>
                     <dfn id="solid-profile">Solid Profile</dfn>
                   </dt>
                   <dd about="#solid-profile" property="skos:definition">
-                    A <dfn>Solid Profile</dfn> is...
+                    A <dfn>Solid Profile </dfn> is the representation of a social agent's identity within the Solid ecosystem, which is used to describe their persona, preferences, abilities, affiliations, social activities and relationships, characteristics, and digital presence, as well as aiding in the discovery of storage and location of specific data.
```
Comment (TallTed) [inline index.html:967]:
```suggestion
                    A <dfn>Solid profile</dfn> is a representation of a social agent's identity within the Solid ecosystem, which can be used to describe their persona, preferences, abilities, affiliations, social activities and relationships, characteristics, and digital presence, as well as to aid in the discovery of storage and location of specific data.
```

File: index.html [hunk 2]
```diff
@@ -982,10 +974,23 @@ <h3 property="schema:name skos:prefLabel">Terminology</h3>
                     property="skos:prefLabel"
                     typeof="skos:Concept"
                   >
-                    <dfn id="extended-profile">Solid Profile</dfn>
+                    <dfn id="extended-profile">Extended Profile</dfn>
```
Comment (TallTed) [inline index.html:977]:
```suggestion
                    <dfn id="extended-profile">Extended Solid Profile</dfn>
```

File: index.html [hunk 3]
```diff
@@ -982,10 +974,23 @@ <h3 property="schema:name skos:prefLabel">Terminology</h3>
...
+                <dl>
+                  <dt
+                    about="#preferences-document"
+                    property="skos:prefLabel"
+                    typeof="skos:Concept"
+                  >
+                    <dfn id="preferences-document">Preferences Document</dfn>
+                  </dt>
+                  <dd about="#preferences-document" property="skos:definition">
+                    A <dfn>preferences document</dfn>  is a specific resource intended to store data which helps authorized applications accommodate the user's preferences, such as settings, personalization preferences, and pointers to the user's private data.
```
Comment (TallTed) [inline index.html:993]: I forget the dfn syntax that lets you have multiple spellings/phrasings, so `extended profile` may be treated as equivalent to `extended Solid profile`... which I think would be optimal.
```suggestion
                    An <dfn>extended Solid profile</dfn> is an additional profile document that provides supplemental description to a Solid profile to help distribute and manage information about a WebID for use in different circumstances. Extended profile documents can be used for various purposes, such as organizing information or limiting access to certain data.
                  </dd>
                </dl>

                <dl>
                  <dt
                    about="#preferences-document"
                    property="skos:prefLabel"
                    typeof="skos:Concept"
                  >
                    <dfn id="preferences-document">Preferences Document</dfn>
                  </dt>
                  <dd about="#preferences-document" property="skos:definition">
                    A <dfn>preferences document</dfn> is a specific resource intended to store data which helps authorized applications accommodate the user's preferences, such as settings, personalization details, and pointers to the user's private data.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

File: index.html [hunk 4]
```diff
@@ -972,7 +964,7 @@ <h3 property="schema:name skos:prefLabel">Terminology</h3>
                     <dfn id="solid-profile">Solid Profile</dfn>
                   </dt>
                   <dd about="#solid-profile" property="skos:definition">
-                    A <dfn>Solid Profile</dfn> is...
+                    A <dfn>Solid profile</dfn> is a representation of a social agent's identity within the Solid ecosystem, which can be used to describe their persona, preferences, abilities, affiliations, social activities and relationships, characteristics, and digital presence, as well as to aid in the discovery of storage and location of specific data.
```
Comment (elf-pavlik) [inline index.html:967]:
```suggestion
                    A <dfn>Solid profile</dfn> is a description of a social agent's identity within the Solid ecosystem, which can be used to describe their persona, preferences, abilities, affiliations, social activities and relationships, characteristics, and digital presence, as well as to aid in the discovery of storage and location of specific data.
```

I think representation might suggest a specific serialization / media-type

**Review by timbl — APPROVED** [review]
LGTM

**Review by TallTed — APPROVED** [review]

**Review by csarven — APPROVED** [review]
Good stuff. I don't have anything to add/subtract (right now :)). PR looks like ready to go.

**Pull request merged**